ui/app/AppLayouts/Wallet/views/NetworkSelectorView.qml

```diff
@@ -0,0 +1,125 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQml 2.15
+import QtQml.Models 2.15
+
+import StatusQ 0.1
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Core.Utils 0.1
+
+import utils 1.0
+
+import "../stores/NetworkSelectPopup"
+import "../controls"
+
+StatusListView {
+    id: root
+    /**
+      Expected model structure:
+        chainName      [string]          - chain long name. e.g. "Ethereum" or "Optimism"
+        chainId        [int]             - chain unique identifier
+        iconUrl        [string]          - SVG icon file. e.g. "network/Network=Ethereum"
+    **/
+    property var flatNetworks
+
+    property bool showIndicator: true
+    property bool multiSelection: false
+    property bool interactive: true
+
+    /**
+        The list selected of chain ids
+        It is a read/write property
+        WARNING: Update the array, not the internal content
+    **/
+    property var selection: ([])
+
+    signal toggleNetwork(var network, int index)
```

Comment:
Yes, you're right! ChainId would probably be sufficient because we have the means to extract the additional info. But I can't say I'm against sending the model item through the signal. Could prove itself handy to not require searching the model again to find the extra data needed to handle the signal.

Although now that I think about it, it's not safe to send the `model` from the delegate anyways. We'll need at least to build another plain JS object out of it.

Reply:
Yeah the delegate might be gone by the time you handle the signal